lib/routes.js

```diff
@@ -220,11 +220,16 @@ module.exports = async function (app, opts) {
       return new MER_ERR_GQL_PERSISTED_QUERY_NOT_FOUND('Unknown query')
     }
 
+    // Create individual contexts for multiple operations, otherwise reference the original context
+    const context = request[kRequestContext].operationsCount > 1
+      ? Object.create(request[kRequestContext])
```

Comment:
`Object.create` is preferred over `Object.assign` in order to preserve the original class and constructor, since Mercurius accepts custom context classes.

Check https://github.com/mercurius-js/mercurius/pull/515 for more information.

Reply:
Just checking, there is no chance that request[kRequestContext] is null or undefined here?